cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -684,6 +693,76 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             ctx.emit(Inst::setcc(cc, dst));
         }
 
+        Opcode::Fcmp => {
+            let condcode = inst_fp_condcode(ctx.data(insn)).unwrap();
+
+            let lhs = input_to_reg(ctx, inputs[0]);
+            let rhs = input_to_reg_mem(ctx, inputs[1]);
+            let dst = output_to_reg(ctx, outputs[0]);
+
+            let input_ty = ctx.input_ty(insn, 0);
+            let op = match input_ty {
+                F32 => SseOpcode::Ucomiss,
+                F64 => SseOpcode::Ucomisd,
+                _ => panic!("Bad input type to Fcmp"),
+            };
+
+            // Unordered is returned by setting ZF, PF, CF <- 111
+            // Greater than by ZF, PF, CF <- 000
+            // Less than by ZF, PF, CF <- 001
+            // Equal by ZF, PF, CF <- 100
+            //
+            // Checking the result of comiss is somewhat annoying because you don't
+            // have setcc instructions that explicitly check simultaneously for the condition
+            // (i.e. eq, le, gt, etc) and orderedness. So that might mean we need more
+            // than one setcc check and then a logical "and" or "or" to determine both.
+            // However knowing that if the parity bit is set, then the result was
+            // considered unordered and knowing that if the parity bit is set, then both
+            // the ZF and CF flag bits must also be set we can getaway with using one setcc
+            // for most condition codes.
+            match condcode {
+                // setb and setbe for ordered LessThan and LessThanOrEqual check if CF = 1 which
+                // doesn't exclude unorderdness. To get around this we can reverse the operands
+                // and the cc test to instead check if CF and ZF are 0 which would also exlucdes
+                // unorderedness. Using similiar logic we also reverse UnorderedOrGreaterThan and
+                // UnorderedOrGreaterThanOrEqual and assure that ZF or CF is 1 to exlude orderedness.
+                FloatCC::LessThan
+                | FloatCC::LessThanOrEqual
+                | FloatCC::UnorderedOrGreaterThan
+                | FloatCC::UnorderedOrGreaterThanOrEqual => {
+                    let lhs = input_to_reg_mem(ctx, inputs[0]);
+                    let rhs = input_to_reg(ctx, inputs[1]);
+                    let dst = output_to_reg(ctx, outputs[0]);
```

Comment:
`dst` is redefined exactly the same way here, so we can get rid of it.

Also, it's not ideal to redefine lhs and rhs here (those `input_to_reg` functions may have side-effects). Instead, can you remove the definition of lhs and rhs that live above the match, and redefine them in each arm, please? (So nothing to do for lhs and rhs here, only redefine them properly in the two other arms.)\

(Unless you find an even better way, of course!)